src/SwapMath.sol

```diff
@@ -37,133 +66,117 @@ library SwapMath {
-            exactIn := iszero(slt(amountRemaining, 0))
-            // mask = 0 if amountRemaining >= 0 else -1
-            let mask := sub(exactIn, 1)
-            amountRemainingAbs := xor(mask, add(mask, amountRemaining))
-        }
-
-        if (exactIn) {
-            uint256 amountRemainingLessFee = FullMath.mulDiv(
-                amountRemainingAbs,
-                UnsafeMath.sub(MAX_FEE_PIPS, feePips),
-                MAX_FEE_PIPS
-            );
-            amountIn = zeroForOne
-                ? SqrtPriceMath.getAmount0Delta(
-                    sqrtRatioTargetX96,
-                    sqrtRatioCurrentX96,
-                    liquidity,
-                    true
-                )
-                : SqrtPriceMath.getAmount1Delta(
-                    sqrtRatioCurrentX96,
-                    sqrtRatioTargetX96,
-                    liquidity,
-                    true
-                );
-            if (amountRemainingLessFee >= amountIn) {
-                // `amountIn` is capped by the target price
-                sqrtRatioNextX96 = sqrtRatioTargetX96;
-            } else {
-                // Exhaust the remaining amount
-                sqrtRatioNextX96 = SqrtPriceMath.getNextSqrtPriceFromInput(
-                    sqrtRatioCurrentX96,
-                    liquidity,
-                    amountRemainingLessFee,
-                    zeroForOne
-                );
-            }
-        } else {
-            amountOut = zeroForOne
-                ? SqrtPriceMath.getAmount1Delta(
-                    sqrtRatioTargetX96,
-                    sqrtRatioCurrentX96,
-                    liquidity,
-                    false
-                )
-                : SqrtPriceMath.getAmount0Delta(
-                    sqrtRatioCurrentX96,
-                    sqrtRatioTargetX96,
-                    liquidity,
-                    false
-                );
-            if (amountRemainingAbs >= amountOut) {
-                // `amountOut` is capped by the target price
-                sqrtRatioNextX96 = sqrtRatioTargetX96;
-            } else {
-                // Exhaust the remaining amount
-                sqrtRatioNextX96 = SqrtPriceMath.getNextSqrtPriceFromOutput(
-                    sqrtRatioCurrentX96,
-                    liquidity,
-                    amountRemainingAbs,
-                    zeroForOne
-                );
+        unchecked {
+            bool zeroForOne = sqrtRatioCurrentX96 >= sqrtRatioTargetX96;
+            uint256 feeComplement = MAX_FEE_PIPS - feePips;
+            bool exactIn;
+            uint256 amountRemainingAbs;
+            assembly {
+                // exactIn = 1 if amountRemaining >= 0 else 0
+                exactIn := iszero(slt(amountRemaining, 0))
+                // mask = 0 if amountRemaining >= 0 else -1
+                let mask := sub(exactIn, 1)
+                amountRemainingAbs := xor(mask, add(mask, amountRemaining))
```

Comment:
Can we use `TernaryLib.abs(...)` here?

Reply:
We can if we want to increase gas. `TernaryLib.abs` contains an `slt` which is a temporary variable also used by `exactIn`.

Reply:
ok.